safe_transaction_service/tokens/services/price_service.py

```diff
@@ -360,3 +360,30 @@ def get_cached_usd_values(
                 FiatCode.USD,
                 token_eth_values_with_timestamp.timestamp,
             )
+
+    def get_eth_price_oracles(self, token_address: ChecksumAddress):
+        """
+        Calculate eth_price from oracles
+        :param token_address
+        """
+        return (
+            self.get_token_eth_value(token_address)
+            or self.get_token_usd_price(token_address)
+            / self.get_native_coin_usd_price()
+        )
+
+    def get_eth_price_composed_oracles(self, token_address: ChecksumAddress):
```

Comment:
I don't feel confident with this function and the unique test that I could see is checking that the result is greater than 0 `self.assertGreater(eth_value_with_timestamp.eth_value, 0.0)` 
Maybe add a new test for this function mocking the composed oracles? What do you think @Uxio0 ?

Reply:
Agree, please add a better test, you can use a well known BALANCER pool like https://app.balancer.fi/#/pool/0xa6f548df93de924d73be7d25dc02554c6bd66db500020000000000000000000e

Reply:
> Agree, please add a better test, you can use a well known BALANCER pool like https://app.balancer.fi/#/pool/0xa6f548df93de924d73be7d25dc02554c6bd66db500020000000000000000000e

Finally I mocked the test because I think that we should test the function that calculates the prices from composed oracles instead test the oracles. Test oracles should be done in safe-eth-py project.